sigh/lib/sigh/runner.rb

```diff
@@ -88,19 +88,34 @@ def fetch_profiles
         includes += ',certificates'
       end
 
+      # also include devices if specific device udids were specified
+      force_profile_device_udids = []
+      unless Sigh.config[:force_profile_device_udids].nil?
+        includes += ",devices"
+        force_profile_device_udids = Sigh.config[:force_profile_device_udids].split(',').map(&:strip)
+      end
+
       results = Sigh.config[:cached_profiles]
       results ||= Spaceship::ConnectAPI::Profile.all(filter: filter, includes: includes)
       results.select! do |profile|
         profile.bundle_id.identifier == Sigh.config[:app_identifier]
       end
 
       results = results.find_all do |current_profile|
+        profile_qualifies = false
         if current_profile.valid? || Sigh.config[:force]
-          true
+          profile_qualifies = true
+          if force_profile_device_udids.length > 0
+            filtered_devices = current_profile.devices.select { |device| force_profile_device_udids.include?(device.udid) }
+            if force_profile_device_udids.length != filtered_devices.length
```

Comment:
are we here checking that the profile contains all the udids?

Maybe something like 
```
if force_profile_device_udids.length > 0
  # all forced udids found in current_profile
  profile_devices = current_profile.devices.map{|d| d.udid}
  unless (force_profile_device_udids - profile_devices).empty?
    UI...
  end
end
```

Reply:
@lacostej this check @ lines 109, 110 is validating that the list of UDIDs in the `current_profile` contains ALL of the UDIDs that are in`force_profile_device_udids`. 

example 1:

current state:
`force_profile_device_udids = [A,B,E,F]`
`current_profile.devices = [A,B,C,D]`

after the `current_profile.devices.select`, `filtered_devices = [A,B]`

`force_profile_device_udids.length (4) != filtered_devices.length (2)` results in `false` so it will be skipped.

Your suggestion above would result in:

`force_profile_device_udids - profile_devices` = `[E,F]` which is not empty so it will be skipped.

Both of these do the same thing, am I missing something with regards to what you are asking for?